app/forms/waste_carriers_engine/payment_summary_form.rb

```diff
@@ -4,18 +4,20 @@ module WasteCarriersEngine
   class PaymentSummaryForm < BaseForm
     attr_accessor :temp_payment_method, :type_change, :registration_cards, :registration_card_charge, :total_charge
 
+    validates :temp_payment_method, inclusion: { in: %w[card bank_transfer] }
+
     def self.can_navigate_flexibly?
       false
     end
 
     def initialize(transient_registration)
       super
-      self.temp_payment_method = @transient_registration.temp_payment_method
+      self.temp_payment_method = transient_registration.temp_payment_method
```

Comment:
NIT Adding a line between super and this as per other changes

```suggestion
      
      self.temp_payment_method = transient_registration.temp_payment_method
```